store/src/java/com/zimbra/cs/imap/ImapServerListener.java

```diff
@@ -255,10 +338,9 @@ private synchronized void processAdminWaitSetResponse(AdminWaitSetResponse wsRes
         }
         if(wsResp.getCanceled() != null && wsResp.getCanceled().booleanValue() && wsID.equalsIgnoreCase(respWSId)) {
             //this waitset was canceled
-            //TODO: figure out what to do with listeners if they are still registered
-            wsID = null;
-            lastSequence.set(0);
-            ZimbraLog.imap.debug("AdminWaitSet %s was canceled", respWSId);
+            ZimbraLog.imap.warn("AdminWaitSet %s was cancelled", respWSId);
```

Comment:
Is this an unexpected situation?  If not then debug logging may make more sense.